content/docs/user-guide/project-structure/pipelines-files.md

```diff
@@ -51,8 +51,17 @@ If it writes files or dirs, they can be defined as <abbr>outputs</abbr>
 ### Parameter dependencies
 
 [Parameters](/doc/command-reference/params) are a special type of stage
-dependency. They consist of a name/value pair to find in a YAML, JSON, TOML, or
-Python parameters file (`params.yaml` by default). Example:
+dependency. They consist of a list of params to track in one of these formats:
+
+1. A param key/value pair that can be found in `params.yaml` (default
+   params file);
+2. A dictionary named by the file path to a custom params file, and with a
+   list of param key/value pairs to find in it;
+3. An empty set (give no value or use `null`) named by the file path to a
+   params file: to track all the params in it dynamically. 
```

Comment:
> The detail given now may be too much but we can address that... (from https://github.com/iterative/dvc.org/pull/3431#pullrequestreview-936901013)

Well... I ended up adding even more details here. This should at some point get extracted to a separate guide (https://github.com/iterative/dvc.org/issues/2572) but OK to keep for now, I think.